js/ui/animation.js

```diff
@@ -66,13 +66,21 @@ var Animation = new Lang.Class({
             newFrameActor.show();
     },
 
+    _convertDelay: function(delay) {
+        if (typeof delay === 'string') {
+            let [delay_a, delay_b] = delay.split('-');
+            return GLib.random_int_range(parseInt(delay_a), parseInt(delay_b));
+        }
+        return delay;
```

Comment:
As I mentioned during our hangout, I don't love the idea of having different types of values mixed together, even if the language allows it. I'd rather have a list if we really need random delays. But just FTR now, I also think that if we need randomly delayed frames, then we should have two animations and specify when to load one of the other (and those could be random).

Reply:
I just meant to express my concerns for the other reviewers to see them, since we talked about it 1:1. I am not picking on your solution, just on the concept that we have random delays for frame animations, because the need for the random case seems to be because we're mixing two animations (blinking + normal). Having the random intervals in JSON or in coding is a different matter, because we could specify the sequence of animations and have the random start times there.
Maybe I should have written this in the Phab task itself. Just so it's clear, this comment is not about invalidating the whole PR, it's about my concern for this complex use-case to satisfy one single animation concept (so far).